# **CREATING INTERACTIVE MAPS**

# 2.1 Introduction

#2.2 Technical Requirements

In [ ]:
import ee
import geemap

import ee: Kode ini mengimpor pustaka ee (singkatan dari Earth Engine). Earth Engine adalah platform komputasi awan dengan katalog multi-petabyte dari citra satelit dan kumpulan data geospasial1. Pustaka ini harus diimpor dan diinisialisasi untuk setiap sesi dan skrip Python baru.

import geemap: Kode ini mengimpor pustaka geemap. Geemap adalah paket Python untuk analisis geospasial dan visualisasi interaktif dengan Google Earth Engine (GEE)2. Dibangun di atas ipyleaflet dan ipywidgets, dan memungkinkan pengguna untuk menganalisis dan memvisualisasikan kumpulan data Earth Engine secara interaktif dalam lingkungan berbasis Jupyter

In [ ]:
geemap.ee_initialize()

To authorize access needed by Earth Engine, open the following URL in a web browser and follow the instructions. If the web browser does not start automatically, please manually browse the URL below.

    https://code.earthengine.google.com/client-auth?scopes=https%3A//www.googleapis.com/auth/earthengine%20https%3A//www.googleapis.com/auth/devstorage.full_control&request_id=Uh_47QJyd1ggFM-VxKs4gqWMii1HMNZvCPgYL4Za2us&tc=SQWSNdmcVqRFOCCPyYvzMqir7RVsdEqgdqWXNAZ95nk&cc=I9GgpxuuAQw-cAs3tEqoPC9-KdKjZFxLyvu9wOzMTMI

The authorization workflow will generate a code, which you should paste in the box below.
Enter verification code: 4/1AfJohXmX8C_q42q831qJtA_7YQvSiSemGS80X2kOaerJgZ3fTYqieqNgETE

Successfully saved authorization token.


Fungsi geemap.ee_initialize() digunakan untuk menginisialisasi Google Earth Engine (GEE) dalam lingkungan Python. Ini adalah langkah penting sebelum Anda dapat menggunakan fungsi dan metode yang disediakan oleh pustaka ee dan geemap

#2.3 Plotting Backends

2.3.1 Ipyleaflet

In [ ]:
import geemap

import geemap: Kode ini mengimpor pustaka geemap. Geemap adalah paket Python untuk analisis geospasial dan visualisasi interaktif dengan Google Earth Engine (GEE)2. Dibangun di atas ipyleaflet dan ipywidgets, dan memungkinkan pengguna untuk menganalisis dan memvisualisasikan kumpulan data Earth Engine secara interaktif dalam lingkungan berbasis Jupyter

In [ ]:
Map = geemap.Map()

Map = geemap.Map() digunakan untuk membuat peta interaktif berbasis ipyleaflet menggunakan pustaka geemap. Peta ini dapat digunakan untuk visualisasi dan analisis data geospasial dari Google Earth Engine dalam lingkungan Jupyter

In [ ]:
Map = geemap.Map(center=[40, -100], zoom=4, height=600)
Map

Map(center=[40, -100], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(ch…

Map = geemap.Map(center=[40, -100], zoom=4, height=600) digunakan untuk membuat peta interaktif dengan pusat di koordinat [40, -100] (lintang 40, bujur -100), tingkat zoom 4, dan tinggi 600 piksel menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

In [ ]:
Map = geemap.Map(data_ctrl=False, toolbar_ctrl=False, draw_ctrl=False)
Map

Map(center=[0, 0], controls=(ZoomControl(options=['position', 'zoom_in_text', 'zoom_in_title', 'zoom_out_text'…

Map = geemap.Map(data_ctrl=False, toolbar_ctrl=False, draw_ctrl=False) digunakan untuk membuat peta interaktif menggunakan pustaka geemap, dengan kontrol data, toolbar, dan kontrol gambar dinonaktifkan.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

In [ ]:
Map = geemap.Map(lite_mode=True)
Map

Map(center=[0, 0], controls=(ZoomControl(options=['position', 'zoom_in_text', 'zoom_in_title', 'zoom_out_text'…

Kode Map = geemap.Map(lite_mode=True) digunakan untuk membuat peta interaktif dengan mode ringan (lite mode) menggunakan pustaka geemap. Dalam mode ringan, hanya kontrol zoom in/out yang ditampilkan pada peta.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

In [ ]:
Map.save('ipyleaflet.html')

Fungsi Map.save('ipyleaflet.html') digunakan untuk menyimpan peta interaktif yang telah dibuat sebagai file HTML dengan nama 'ipyleaflet.html’.
Setelah menjalankan kode ini, peta interaktif akan disimpan sebagai file HTML ‘ipyleaflet.html’ di direktori kerja saat ini. file dapat dibuka di browser web untuk melihat peta.

2.3.2 Folium

In [ ]:
import geemap.foliumap as geemap

Kode import geemap.foliumap as geemap digunakan untuk mengimpor modul foliumap dari pustaka geemap. Modul foliumap ini memungkinkan Anda membuat peta interaktif berbasis folium.


In [ ]:
Map = geemap.Map(center=[40, -100], zoom=4, height=600)
Map

Kode Map = geemap.Map(center=[40, -100], zoom=4, height=600) digunakan untuk membuat peta interaktif dengan pusat di koordinat [40, -100] (lintang 40, bujur -100), tingkat zoom 4, dan tinggi 600 piksel menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

In [ ]:
Map.save('folium.html')

Fungsi Map.save('folium.html') digunakan untuk menyimpan peta interaktif yang telah Anda buat sebagai file HTML dengan nama 'folium.html’.

2.3.3 Plotly

In [ ]:
import geemap.plotlymap as geemap

Kode import geemap.plotlymap as geemap digunakan untuk mengimpor modul plotlymap dari pustaka geemap. Modul plotlymap ini memungkinkan membuat peta interaktif berbasis plotly.

In [ ]:
Map = geemap.Map()
Map

Kode Map = geemap.Map() digunakan untuk membuat peta interaktif menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

In [ ]:
from google.colab import output
output.enable_custom_widget_manager()

Support for third party widgets will remain active for the duration of the session. To disable support:

In [ ]:
from google.colab import output
output.disable_custom_widget_manager()

2.3.4 Pydeck

In [ ]:
pip install pydeck

In [ ]:
import geemap.deck as geemap

Kode import geemap.deck as geemap digunakan untuk mengimpor modul deck dari pustaka geemap. Modul deck ini memungkinkan Anda untuk berinteraksi dengan deck.gl dalam Python

In [ ]:
Map = geemap.Map()
Map

<IPython.core.display.Javascript object>

{
  "initialViewState": {
    "height": 800,
    "latitude": 20,
    "longitude": 0,
    "zoom": 1.2
  },
  "layers": [],
  "mapProvider": "carto",
  "mapStyle": "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
  "views": [
    {
      "@@type": "MapView",
      "controller": true
    }
  ]
}

Kode Map = geemap.Map() digunakan untuk membuat peta interaktif menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

2.3.5 Kepler GL

In [ ]:
pip install keplergl

In [ ]:
import geemap.kepler as geemap

Kode import geemap.kepler as geemap digunakan untuk mengimpor modul kepler dari pustaka geemap. Modul kepler ini memungkinkan untuk berinteraksi dengan kepler.gl dalam Python

In [ ]:
Map = geemap.Map()
Map

Map(config={'version': 'v1', 'config': {'mapState': {'latitude': 20, 'longitude': 0, 'zoom': 1.3, 'bearing': 0…

Kode Map = geemap.Map() digunakan untuk membuat peta interaktif menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

#2.4 Adding basemaps

2.4.1 Built-in basemaps

In [ ]:
import geemap

Kode import geemap digunakan untuk mengimpor pustaka geemap dalam Python1geemap adalah paket Python untuk analisis geospasial dan visualisasi interaktif dengan Google Earth Engine (GEE). Dibangun di atas ipyleaflet dan ipywidgets, dan memungkinkan pengguna untuk menganalisis dan memvisualisasikan kumpulan data Earth Engine secara interaktif dalam lingkungan berbasis Jupyter

In [ ]:
Map = geemap.Map(basemap='HYBRID')
Map

Map(center=[0, 0], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(childr…

Kode Map = geemap.Map(basemap='HYBRID') digunakan untuk membuat peta interaktif menggunakan pustaka geemap dengan basemap (peta dasar) diatur ke 'HYBRID’. Basemap ‘HYBRID’ biasanya menampilkan peta satelit dengan label jalan dan landmark

In [ ]:
Map.add_basemap('OpenTopoMap')

Fungsi Map.add_basemap('OpenTopoMap') digunakan untuk menambahkan basemap (peta dasar) ‘OpenTopoMap’ ke peta interaktif yang telah dibuat menggunakan pustaka geemap. Setelah menjalankan kode ini, basemap ‘OpenTopoMap’ akan ditambahkan ke peta


In [ ]:
for basemap in geemap.basemaps.keys():
 print(basemap)

OpenStreetMap
Esri.WorldStreetMap
Esri.WorldImagery
Esri.WorldTopoMap
FWS NWI Wetlands
FWS NWI Wetlands Raster
NLCD 2021 CONUS Land Cover
NLCD 2019 CONUS Land Cover
NLCD 2016 CONUS Land Cover
NLCD 2013 CONUS Land Cover
NLCD 2011 CONUS Land Cover
NLCD 2008 CONUS Land Cover
NLCD 2006 CONUS Land Cover
NLCD 2004 CONUS Land Cover
NLCD 2001 CONUS Land Cover
USGS NAIP Imagery
USGS NAIP Imagery False Color
USGS NAIP Imagery NDVI
USGS Hydrography
USGS 3DEP Elevation
ESA Worldcover 2020
ESA Worldcover 2020 S2 FCC
ESA Worldcover 2020 S2 TCC
ESA Worldcover 2021
ESA Worldcover 2021 S2 FCC
ESA Worldcover 2021 S2 TCC
BasemapAT.basemap
BasemapAT.grau
BasemapAT.highdpi
BasemapAT.orthofoto
BasemapAT.overlay
BasemapAT.surface
BasemapAT.terrain
CartoDB.DarkMatter
CartoDB.DarkMatterNoLabels
CartoDB.DarkMatterOnlyLabels
CartoDB.Positron
CartoDB.PositronNoLabels
CartoDB.PositronOnlyLabels
CartoDB.Voyager
CartoDB.VoyagerLabelsUnder
CartoDB.VoyagerNoLabels
CartoDB.VoyagerOnlyLabels
CyclOSM
Esri.AntarcticBasema

Kode for basemap in geemap.basemaps.keys(): print(basemap) digunakan untuk mencetak semua basemap (peta dasar) yang tersedia dalam pustaka geemap

In [ ]:
len(geemap.basemaps)

143

Fungsi len(geemap.basemaps) dalam Python digunakan untuk menghitung jumlah basemap (peta dasar) yang tersedia dalam pustaka geemap.

Terdapat 143 peta dasar yang tersedia dalam pustaka geemap.

2.4.2. XYZ tiles

In [ ]:
Map = geemap.Map(center=[40, -100],
                 url="https://mt1.google.com/vt/lyrs=p&x={x}&y={y}&z={z}",
                 name="Google Terrain",
                 attribution="Google"
                 )

Map

Map(center=[40, -100], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(ch…

Kode Map = geemap.Map(center=[40, -100], url="https://mt1.google.com/vt/lyrs=p&x={x}&y={y}&z={z}", name="Google Terrain", attribution="Google") digunakan untuk membuat peta interaktif dengan pusat di koordinat [40, -100] (lintang 40, bujur -100), dan menggunakan basemap (peta dasar) khusus dari URL yang diberikan

2.4.3 WMS tiles

In [ ]:
Map = geemap.Map(center=[40, -100], zoom=4)
url = 'https://www.mrlc.gov/geoserver/mrlc_display/NLCD_2019_Land_Cover_L48/wms?'
Map.add_wms_layer(
    url=url,
    layers='NLCD_2019_Land_Cover_L48',
    name='NLCD 2019',
    format='image/png',
    attribution='MRLC',
    transparent=True,
    )

Map

Map(center=[40, -100], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(ch…

Kode Map = geemap.Map(center=[40, -100], url="https://mt1.google.com/vt/lyrs=p&x={x}&y={y}&z={z}", name="Google Terrain", attribution="Google") digunakan untuk membuat peta interaktif dengan pusat di koordinat [40, -100] (lintang 40, bujur -100), dan menggunakan basemap (peta dasar) khusus dari URL yang diberikan.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

2.4.4 Planet Basemaps

In [ ]:
import os
os.environ["PLANET_API_KEY"] = "YOUR_API_KEY"

Kode os.environ["PLANET_API_KEY"] = "YOUR_API_KEY" digunakan untuk mengatur variabel lingkungan PLANET_API_KEY ke nilai "YOUR_API_KEY" dalam Python. Variabel lingkungan ini biasanya digunakan untuk menyimpan kunci API Planet, yang diperlukan untuk mengakses API Planet.

In [ ]:
quarterly_tiles = geemap.planet_quarterly_tiles()
for tile in quarterly_tiles:
 print(tile)

Planet_2016q1
Planet_2016q2
Planet_2016q3
Planet_2016q4
Planet_2017q1
Planet_2017q2
Planet_2017q3
Planet_2017q4
Planet_2018q1
Planet_2018q2
Planet_2018q3
Planet_2018q4
Planet_2019q1
Planet_2019q2
Planet_2019q3
Planet_2019q4
Planet_2020q1
Planet_2020q2
Planet_2020q3
Planet_2020q4
Planet_2021q1
Planet_2021q2
Planet_2021q3
Planet_2021q4
Planet_2022q1
Planet_2022q2
Planet_2022q3
Planet_2022q4
Planet_2023q1
Planet_2023q2
Planet_2023q3


Kode quarterly_tiles = geemap.planet_quarterly_tiles(); for tile in quarterly_tiles: print(tile) digunakan untuk mendapatkan daftar semua tile Planet yang tersedia per kuartal dan mencetaknya.

In [ ]:
monthly_tiles = geemap.planet_monthly_tiles()
for tile in monthly_tiles:
 print(tile)

Planet_2016_01
Planet_2016_02
Planet_2016_03
Planet_2016_04
Planet_2016_05
Planet_2016_06
Planet_2016_07
Planet_2016_08
Planet_2016_09
Planet_2016_10
Planet_2016_11
Planet_2016_12
Planet_2017_01
Planet_2017_02
Planet_2017_03
Planet_2017_04
Planet_2017_05
Planet_2017_06
Planet_2017_07
Planet_2017_08
Planet_2017_09
Planet_2017_10
Planet_2017_11
Planet_2017_12
Planet_2018_01
Planet_2018_02
Planet_2018_03
Planet_2018_04
Planet_2018_05
Planet_2018_06
Planet_2018_07
Planet_2018_08
Planet_2018_09
Planet_2018_10
Planet_2018_11
Planet_2018_12
Planet_2019_01
Planet_2019_02
Planet_2019_03
Planet_2019_04
Planet_2019_05
Planet_2019_06
Planet_2019_07
Planet_2019_08
Planet_2019_09
Planet_2019_10
Planet_2019_11
Planet_2019_12
Planet_2020_01
Planet_2020_02
Planet_2020_03
Planet_2020_04
Planet_2020_05
Planet_2020_06
Planet_2020_07
Planet_2020_08
Planet_2020_09
Planet_2020_10
Planet_2020_11
Planet_2020_12
Planet_2021_01
Planet_2021_02
Planet_2021_03
Planet_2021_04
Planet_2021_05
Planet_2021_06
Planet_202

Kode monthly_tiles = geemap.planet_monthly_tiles(); for tile in monthly_tiles: print(tile) digunakan untuk mendapatkan daftar semua tile Planet yang tersedia per bulan dan mencetaknya.

In [ ]:
Map = geemap.Map()
Map.add_planet_by_month(year=2020, month=8)
Map


Map(center=[0, 0], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(childr…

Kode Map = geemap.Map(); Map.add_planet_by_month(year=2020, month=8); Map digunakan untuk membuat peta interaktif menggunakan pustaka geemap, menambahkan basemap (peta dasar) Planet untuk bulan Agustus 2020, dan menampilkan peta.

In [ ]:
Map

Map(center=[0, 0], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(childr…

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.

2.4.5 Basemap GUI

In [ ]:
import os
os.environ["PLANET_API_KEY"] = "YOUR_API_KEY"

Kode os.environ["PLANET_API_KEY"] = "YOUR_API_KEY" digunakan untuk mengatur variabel lingkungan PLANET_API_KEY ke nilai "YOUR_API_KEY" dalam Python. Variabel lingkungan ini biasanya digunakan untuk menyimpan kunci API Planet, yang diperlukan untuk mengakses API Planet.

In [ ]:
Map = geemap.Map()
Map

Map(center=[0, 0], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(childr…

Kode Map = geemap.Map() digunakan untuk membuat peta interaktif menggunakan pustaka geemap.

Kode Map pada baris terakhir digunakan untuk menampilkan peta yang telah dibuat.